sprintf.c

```diff
@@ -1173,6 +1174,7 @@ rb_enc_vsprintf(rb_encoding *enc, const char *fmt, va_list ap)
     BSD_vfprintf(&f, fmt, ap);
     RBASIC_SET_CLASS_RAW(result, rb_cString);
     rb_str_resize(result, (char *)f._p - RSTRING_PTR(result));
+    ENC_CODERANGE_CLEAR(result);
```

Comment:
These two `ENC_CODERANGE_CLEAR` are a bit unfortunate, but these two formatting function were relying on `rb_str_resize` to clear the coderange for them, otherwise they produce strings with incorrect coderange.